src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/PackageFeedEnumerator.cs

```diff
@@ -72,6 +74,15 @@ private PackageFeedEnumerator(PackageFeedEnumerator other)
 
         public void Dispose()
         {
+            if (_isDisposed)
+            {
+                return;
+            }
+
+            _current?.Dispose();
+            GC.SuppressFinalize(this);
```

Comment:
Are we doing this GC suppression consistently? I see it sometimes not all time.

Reply:
The dispose pattern recommends calling `GC.SuppressFinalize` method. We have analyzers that flag if the dispose pattern is not implemented properly. As of now, we addressed only [2/4 issues related to `dispose` pattern](https://github.com/NuGet/Client.Engineering/issues?q=is%3Aissue+dispose+assignee%3Akartheekp-ms). Once we fix all the open issue then there will be consistency.

Reply:
TIL that calling `GC.SuppressFinalize` in `Dispose()` method is optional if the declaring type doesn't have a finalizer method.